core/src/components/searchbar/searchbar.ionic.scss

```
@@ -92,6 +100,16 @@
   opacity: 1;
 }
 
+// Searchbar Search Icon & Cancel Icon
+// -----------------------------------------
+:host(:not(.searchbar-has-leading-icons)) {
+  --leading-icon-padding: #{globals.$ion-scale-0};
+}
+
+:host(:not(.searchbar-has-value)) {
+  --icon-color: #{globals.$ion-icon-subtlest};
+}
```

As one of our standards, we try to not use `:not` when possible. I would recommend having the default under `:host` as `--icon-color: #{globals.$ion-icon-subtlest};` as I mentioned in my earlier comment.

And then update this to be:
```suggestion
:host(.searchbar-has-value) {
  --icon-color: #{globals.$ion-icon-default};
}
```